src/BIP44CoinTypeNode.ts

```diff
@@ -106,45 +85,71 @@ export class BIP44CoinTypeNode implements BIP44CoinTypeNodeInterface {
 
       validateCoinTypeNodeDepth(nodeOrPathTuple.length - 1);
 
-      this.#node = new BIP44Node({
+      const node = await BIP44Node.create({
         derivationPath: nodeOrPathTuple,
       });
 
       // Split the bip32 string token and extract the coin_type index
-      this.coin_type = Number.parseInt(
+      const coinType = Number.parseInt(
         nodeOrPathTuple[BIP_44_COIN_TYPE_DEPTH].split(':')[1].replace(`'`, ''),
         10,
       );
-    } else {
-      validateCoinTypeNodeDepth(nodeOrPathTuple.depth);
-      validateCoinTypeParentKey(nodeOrPathTuple.key);
-
-      const keyBuffer =
-        nodeOrPathTuple instanceof BIP44Node
-          ? nodeOrPathTuple.keyBuffer
-          : base64StringToBuffer(nodeOrPathTuple.key);
-
-      if (
-        typeof coin_type !== 'number' ||
-        !Number.isInteger(coin_type) ||
-        coin_type < 0
-      ) {
-        throw new Error(
-          'Invalid coin type: The specified coin type must be a non-negative integer number.',
-        );
-      }
-      this.coin_type = coin_type;
-
-      this.#node =
-        nodeOrPathTuple instanceof BIP44Node
-          ? nodeOrPathTuple
-          : new BIP44Node({
-              depth: BIP_44_COIN_TYPE_DEPTH,
-              key: keyBuffer,
-            });
+
+      return new BIP44CoinTypeNode(node, coinType);
+    }
+
+    validateCoinTypeNodeDepth(nodeOrPathTuple.depth);
+    validateCoinTypeParentKey(nodeOrPathTuple.key);
+
+    const keyBuffer =
+      nodeOrPathTuple instanceof BIP44Node
+        ? nodeOrPathTuple.keyBuffer
+        : base64StringToBuffer(nodeOrPathTuple.key);
+
+    if (
+      typeof coin_type !== 'number' ||
+      !Number.isInteger(coin_type) ||
+      coin_type < 0
+    ) {
+      throw new Error(
+        'Invalid coin type: The specified coin type must be a non-negative integer number.',
+      );
     }
 
-    this.path = getBIP44CoinTypePathString(this.coin_type);
+    const node =
+      nodeOrPathTuple instanceof BIP44Node
+        ? nodeOrPathTuple
+        : await BIP44Node.create({
+            depth: BIP_44_COIN_TYPE_DEPTH,
+            key: keyBuffer,
+          });
+
+    return new BIP44CoinTypeNode(node, coin_type);
+  }
+
+  readonly #node: BIP44Node;
+
+  public get depth(): BIP44Depth {
+    return this.#node.depth;
+  }
+
+  public get key(): string {
+    return this.#node.key;
+  }
+
+  public get keyBuffer(): Buffer {
+    return this.#node.keyBuffer;
+  }
+
+  public readonly path: CoinTypeHDPathString;
+
+  public readonly coin_type: number;
```

Comment:
Any particular reason not to keep these at the top of the class declaration?

Reply:
Ah, because static methods can't access instance fields anyway?

Reply:
Yeah. Generally I keep static methods before everything else, but if there's another convention we use I'd be happy to change it. Might be something we want to enforce with ESLint too.